Example_Apps/BugSplatTest-macOS-Tool-CPlusPlus/REAME.md

```diff
@@ -0,0 +1,95 @@
+
+        // Initialize BugSplat
+        MyBugSplatDelegate *delegate = [[MyBugSplatDelegate alloc] init];
+
+        // Set a BugSplatDelegate
+        [[BugSplat shared] setDelegate:delegate];
+
+        // Command Line Tools do not have a GUI. setAutoSubmitCrashReport: YES
+        [[BugSplat shared] setAutoSubmitCrashReport:YES];
+
+        NSString *databaseName = [NSString stringWithCString:bugSplatDatabase encoding:NSUTF8StringEncoding];
+        NSLog(@"bugSplatInit called with database name: %@", databaseName);
+
+        // Set the BugSplatDatabase name before calling start
+        [[BugSplat shared] setBugSplatDatabase:databaseName];
+
+        // Don't forget to call start after you've finished configuring BugSplat
+        [[BugSplat shared] start];
+
+        // BugSplat expects a NSApplicationDidFinishLaunchingNotification to be sent before it will process crash reports.
+        // In a command line tool without a normal app launch, send the notification manually. This should be sent after [[BugSplat shared] start]
+        [[NSNotificationCenter defaultCenter] postNotificationName:NSApplicationDidFinishLaunchingNotification object:nil userInfo:[NSDictionary new]];
+    }
+
+    return 0;
+}
+
+```
+
+### Integrating BugSplat.xcframework into Command Line Applications
+
+A command line application can be written in several different languages. BugSplatTest-macOS-Tool-CPlusPlus example illustrates a command line application written in C++. Regardless of what language the command line application is written in, there are a few things that are requirements for successfully integrating BugSplat.xcframework and posting crash reports. These techniques are used in the example app BugSplatTest-macOS-Tool-CPlusPlus. These requirements are additional/modifications to requirements already imposed by a GUI based app. For example, it is still a requirement to use symbol-upload-macos as part of a Build Phase Script. That requirement does not change. Neither does having an app database name. This section highlights key differences from the GUI app requirements.
+
+#### BugSplat expectations regardless of the type of application built:
++ BugSplat needs a database name to post crash reports to
++ BugSplat expects an Info.plist
++ BugSplat expects several values within the Info.plist: 
+1. set Product Bundle Identifier
+2. set Product Name
+3. set Current Project Version
+4. set Marketing Version (optional)
+
++ BugSplat expects an AppKit NSApplicationDidFinishLaunchingNotification NSNotication to be posted when the application has completed the startup phase. In this example, it is programmatically sent after [[BugSplat shaerd] start] is called. This NSNotification allows BugSplat to begin processing any crash reports that may have been captured during a prior application session. This would normally be sent by AppKit or UIKit as part of the normal application life cycle, but it is not sent in a command line application.
```

Comment:
```suggestion
+ BugSplat expects an AppKit NSApplicationDidFinishLaunchingNotification NSNotication to be posted when the application has completed the startup phase. In this example, it is programmatically sent after [[BugSplat shared] start] is called. This NSNotification allows BugSplat to begin processing any crash reports that may have been captured during a prior application session. This would normally be sent by AppKit or UIKit as part of the normal application life cycle, but it is not sent in a command line application.
```